test/e2e/clusterctl_upgrade_test.go

```diff
@@ -207,69 +207,57 @@ var _ = Describe("When testing clusterctl upgrades (v1.0=>current)", func() {
 	})
 })
 
-var _ = Describe("When testing clusterctl upgrades (v1.5=>current)", func() {
-	// Get v1.5 latest stable release
-	version := "1.5"
+var _ = Describe("When testing clusterctl upgrades (v1.6=>current)", func() {
+	// Get v1.6 latest stable release
+	version := "1.6"
 	stableRelease, err := GetStableReleaseOfMinor(ctx, version)
 	Expect(err).ToNot(HaveOccurred(), "Failed to get stable version for minor release : %s", version)
 	ClusterctlUpgradeSpec(ctx, func() ClusterctlUpgradeSpecInput {
 		return ClusterctlUpgradeSpecInput{
-			E2EConfig:              e2eConfig,
```

Comment:
We have to pin this test to 1.6 (i.e. we still need all the lines that have been removed here to pin to 1.6)

Currently this test is installing 1.7: 
```yaml
STEP: Downloading clusterctl binary from https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.4/clusterctl-linux-amd64 - /home/prow/go/src/sigs.k8s.io/cluster-api/test/e2e/common.go:48 @ 05/13/24 10:38:32.244
STEP: Initializing the workload cluster with older versions of providers - /home/prow/go/src/sigs.k8s.io/cluster-api/test/e2e/clusterctl_upgrade.go:314 @ 05/13/24 10:38:32.82
INFO: clusterctl init --config /logs/artifacts/repository/clusterctl-config.yaml --kubeconfig /tmp/e2e-kubeconfig2911029778 --wait-providers --core cluster-api:v1.7.1 --bootstrap kubeadm:v1.7.1 --control-plane kubeadm:v1.7.1 --infrastructure docker:v1.7.1
```

Reply:
Same for the next test: When testing clusterctl upgrades using ClusterClass (v1.6=>current) [ClusterClass]

If we don't set InitWithCoreProvider, ... CAPI will just install a newer version from docker.yaml (1.7)